src/services/codefixes/fixForgottenThisPropertyAccess.ts

```diff
@@ -2,9 +2,11 @@
 namespace ts.codefix {
     const fixId = "forgottenThisPropertyAccess";
     const didYouMeanStaticMemberCode = Diagnostics.Cannot_find_name_0_Did_you_mean_the_static_member_1_0.code;
+    const duplicateIdentifierCode = Diagnostics.Duplicate_identifier_0.code;

```

Comment:
why do private identifiers give "Duplicate identifier"? I wouldn't expect a reference to act like a declaration.

Reply:
Because a lone private identifier is not an expression, it parses as a declaration. I looked at changing the parsing to be more tolerant and issue a grammar error, but it got weird really fast. Once you start going down `parseExpression`, you have to return _some_ valid expression, even if it includes missing tokens/nodes. So it was unclear what to produce that wouldn’t blow up other stuff.